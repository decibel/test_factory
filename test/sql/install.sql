\set ECHO none
\i test/helpers/setup.sql

SET client_min_messages = WARNING;

/*
 * DO NOT use CASCADE here; we want this to fail if there's anything installed
 * that depends on it.
 */
SELECT lives_ok($$DROP EXTENSION IF EXISTS test_factory_pgtap$$, 'drop extension test_factory_pgtap');
SELECT lives_ok($$DROP EXTENSION IF EXISTS test_factory$$, 'drop extension test_factory');

SELECT hasnt_extension( 'test_factory' );
SELECT hasnt_extension( 'test_factory_pgtap' );

SELECT lives_ok($$CREATE EXTENSION test_factory_pgtap CASCADE$$, 'create extension');
COMMIT;

SELECT has_function('tf', 'tap', array['text','text']);

-- Cleanup
SELECT lives_ok($$DROP EXTENSION IF EXISTS test_factory_pgtap$$, 'clean-up test_factory_pgtap');
SELECT lives_ok($$DROP EXTENSION IF EXISTS test_factory$$, 'clean-up test_factory');

/*
 * Arguably we should cleanup pgtap and the tap schema...
 */

-- vi: expandtab ts=2 sw=2

