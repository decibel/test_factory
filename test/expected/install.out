\set ECHO none
ok 1 - drop extension test_factory_pgtap
ok 2 - drop extension test_factory
ok 3 - Extension test_factory should not exist
ok 4 - Extension test_factory_pgtap should not exist
ok 5 - create extension
ok 6 - Function tf.tap(text, text) should exist
ok 7 - clean-up test_factory_pgtap
ok 8 - clean-up test_factory
